pkg/deploy/deploy.go

```diff
@@ -233,8 +233,15 @@ func manageResource(ctx context.Context, cli client.Client, obj *unstructured.Un
 			}
 		}
 
-		if obj.GetOwnerReferences() == nil {
+		existingOwnerReferences := obj.GetOwnerReferences()
```

Comment:
Perhaps we should use updated `obj` here?

```suggestion
		existingOwnerReferences := found.GetOwnerReferences()
```